libs/state/src/lib/rx-state.service.ts

```diff
@@ -53,8 +53,10 @@ export type ProjectValueReducer<T, K extends keyof T, V> = (
 @Injectable()
 export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
   private subscription = new Subscription();
-
-  private accumulator = createAccumulationObservable<T>();
+  private handleError = inject(ErrorHandler).handleError;
+  private accumulator = createAccumulationObservable<T>({
+    handleError: this.handleError,
+  });
```

Comment:
We have to make sure the correct execution context is provided. Not sure if the current solution works if you try to access fields inside the `handleError` function.

e.g.

```ts
class ErrorHandler {
  handleError() {
    if (this.env === 'prod') {
       // do something in prod
    }
  }
}

```

could we create an test case for that?

```suggestion
  private errorHandler = inject(ErrorHandler);
  private accumulator = createAccumulationObservable<T>({
    handleError: this.erroHandler.handleError.bind(this.errorHandler),
  });
```

Reply:
I pushed the changes, but the test for checking that bind works is failing, I'm not sure to get why. If you have an idea @hoebbelsB.